spark/tests/test_spark.py

```diff
@@ -213,239 +220,209 @@ def yarn_requests_auth_mock(*args, **kwargs):
+            return MockedResponse(body, 200)
 
     elif arg_url == STANDALONE_APP_HTML_URL:
         with open(os.path.join(FIXTURE_DIR, 'spark_standalone_app'), 'rb') as f:
             body = f.read()
-            return MockStandaloneResponse(body, 200)
+            return MockedResponse(body, 200)
 
     elif arg_url == STANDALONE_SPARK_APP_URL:
         with open(os.path.join(FIXTURE_DIR, 'spark_apps_pre20'), 'rb') as f:
             body = f.read()
-            return MockStandaloneResponse(body, 200)
+            return MockedResponse(body, 200)
 
     elif arg_url == STANDALONE_SPARK_JOB_URL:
-        return MockStandaloneResponse("{}", 404)
+        return MockedResponse("{}", 404)
 
     elif arg_url == STANDALONE_SPARK_STAGE_URL:
-        return MockStandaloneResponse("{}", 404)
+        return MockedResponse("{}", 404)
 
     elif arg_url == STANDALONE_SPARK_EXECUTOR_URL:
-        return MockStandaloneResponse("{}", 404)
+        return MockedResponse("{}", 404)
 
     elif arg_url == STANDALONE_SPARK_RDD_URL:
-        return MockStandaloneResponse("{}", 404)
+        return MockedResponse("{}", 404)
 
     elif arg_url == STANDALONE_SPARK_STREAMING_STATISTICS_URL:
-        return MockStandaloneResponse("{}", 404)
+        return MockedResponse("{}", 404)
 
     elif arg_url == STANDALONE_SPARK_JOB_URL_PRE20:
         with open(os.path.join(FIXTURE_DIR, 'job_metrics'), 'rb') as f:
             body = f.read()
-            return MockStandaloneResponse(body, 200)
+            return MockedResponse(body, 200)
 
     elif arg_url == STANDALONE_SPARK_STAGE_URL_PRE20:
         with open(os.path.join(FIXTURE_DIR, 'stage_metrics'), 'rb') as f:
             body = f.read()
-            return MockStandaloneResponse(body, 200)
+            return MockedResponse(body, 200)
 
     elif arg_url == STANDALONE_SPARK_EXECUTOR_URL_PRE20:
         with open(os.path.join(FIXTURE_DIR, 'executor_metrics'), 'rb') as f:
             body = f.read()
-            return MockStandaloneResponse(body, 200)
+            return MockedResponse(body, 200)
 
     elif arg_url == STANDALONE_SPARK_RDD_URL_PRE20:
         with open(os.path.join(FIXTURE_DIR, 'rdd_metrics'), 'rb') as f:
             body = f.read()
-            return MockStandaloneResponse(body, 200)
+            return MockedResponse(body, 200)
 
     elif arg_url == STANDALONE_SPARK_STREAMING_STATISTICS_URL_PRE20:
         with open(os.path.join(FIXTURE_DIR, 'streaming_statistics'), 'rb') as f:
             body = f.read()
-            return MockStandaloneResponse(body, 200)
+            return MockedResponse(body, 200)
 
     elif arg_url == VERSION_PATH:
         with open(os.path.join(FIXTURE_DIR, 'version'), 'rb') as f:
             body = f.read()
-            return MockStandaloneResponse(body, 200)
+            return MockedResponse(body, 200)
+
+
+def proxy_with_warning_page_mock(session, *args, **kwargs):
+    session_cookie = session.cookies.get('proxy_cookie')
+    arg_url = args[0]
```

Comment:
Nit: we can use `proxy_with_warning_page_mock(session, url, *args, **kwargs)` to not have to infer the URL argument.